tests/ui/unnecessary_cast.stderr

```diff
@@ -1,190 +1,232 @@
+error: casting raw pointers to the same type and constness is unnecessary (`*const T` -> `*const T`)
```

Comment:
Wouldn't the type inside the bracket always be equal? i.e.

(`foo` -> `bar`) always will be the case that `foo==bar` in this lint?

Reply:
What do you think of removing this altogether? Seems redundant

Reply:
That was done to remain consistent with the other lint message further down (`&format!("casting to the same type is unnecessary ({cast_from} -> {cast_to})")`). It is redundant yes but imo both should be changed if it's removed

Reply:
That's fair. We can leave it for now